projects/observability/src/shared/components/explore-query-editor/explore-visualization-builder.ts

```diff
@@ -234,6 +234,21 @@ export class ExploreVisualizationBuilder implements OnDestroy {
 
     return of(interval);
   }
+
+  private encodeGroupBy(groupBy: GraphQlGroupBy): GraphQlGroupBy {
+    return {
+      ...groupBy,
+      keyExpressions:
+        groupBy.keyExpressions.length > 0
+          ? [
+              {
+                ...groupBy.keyExpressions[0],
+                subpath: camelCase(groupBy.keyExpressions[0].subpath)
```

Comment:
Is the only goal here to camel case this?
1. If this needs to be done, this isn't the place to do it. It should be where the group by expression is built, so we don't have two different objects with the same type one valid, one invalid.
2. Why does it need to be done in the first place? Subpaths are user-defined schemas, so if they gave us something with a particular casing, we should be respecting that.

Reply:
@aaron-steinfeld the issue here appears when we try to use the hyphened result as a variable. So I took this strategy as a workaround. The particular issue appears on the selection, is not possible to use it as a variable (the same for other types of special characters like `'/'` or `'.'`. I understand probably this would be not the best fix but I'm not able to find a better solution. 

![image](https://user-images.githubusercontent.com/79482271/187230599-cf7a76f7-119f-4118-b983-d794e4dfabe9.png)

Reply:
Discussed offline, the issue is the alias can't have the same special chars a subpath supports, so the appropriate fix would be to update the spec builder which generates the alias given an expression to replace those special chars and leave the subpath untouched (it already does this with some special chars, but not `-`)

Reply:
done